src/odemis/acq/stream/test/stream_helper_test.py

```
@@ -0,0 +1,349 @@
+        self.assertAlmostEqual(eratio, rep_ratio, places=1)
+
+        # When asking for a square area, the ROI compensates for the non-square eshape
+        helper.repetition.value = (1, 1)
+        roi = helper.roi.value
+        roi_size = (roi[2] - roi[0], roi[3] - roi[1])
+        roi_ratio = roi_size[0] / roi_size[1]
+        self.assertAlmostEqual(eratio, 1 / roi_ratio, places=2)
+
+        # The following changes would cause the RoI[1] to be negative (by a very tiny value).
+        # => Make sure the values are always strictly within 0 and 1.
+        e_fov = comp.compute_scanner_fov(self.ebeam)
```

Ruff (0.15.12)

[error] Undefined name `comp`

(F821)